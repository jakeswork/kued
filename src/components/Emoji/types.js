import { shape, node, string } from 'prop-types';

export const propTypes = {
  classes: shape({}).isRequired,
  emoji: node,
  label: string.isRequired,
};

export const defaultProps = {
  emoji: null,
};
