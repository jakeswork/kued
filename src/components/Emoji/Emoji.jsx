import React from 'react';

import { propTypes, defaultProps } from './types';

const Emoji = ({
  classes,
  label,
  icon,
  ...props
}) => (
  <span role="img" aria-label={label} className={classes.root} {...props}>
    { icon }
  </span>
);

Emoji.propTypes = propTypes;

Emoji.defaultProps = defaultProps;

export default Emoji;
