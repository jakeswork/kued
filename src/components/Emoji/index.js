import withStyles from 'react-jss';

import Emoji from './Emoji';
import styles from './styles';

export default withStyles(styles)(Emoji);
