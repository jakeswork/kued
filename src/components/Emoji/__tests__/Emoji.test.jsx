import React from 'react';
import { shallow } from 'enzyme';

import Emoji from '../Emoji';

const defaultProps = {
  classes: {},
  label: 'test',
};

describe('The Emoji instance', () => {
  describe('When called with default props', () => {
    it('Should render without throwing an error', () => {
      const wrapper = shallow(<Emoji {...defaultProps} />);

      expect(wrapper).toMatchSnapshot();
    });
  });
});
