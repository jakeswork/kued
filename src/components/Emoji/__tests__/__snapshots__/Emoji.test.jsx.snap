// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Emoji instance When called with default props Should render without throwing an error 1`] = `
<span
  aria-label="test"
  emoji={null}
  role="img"
/>
`;
